cmd/knit/cmd/root.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ */
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+)
+
+type knitOptions struct {
+	cpuList    string
+	cpus       cpuset.CPUSet
+	procFSRoot string
+	sysFSRoot  string
+}
+
+var knitOpts knitOptions
+
+// NewRootCommand returns entrypoint command to interact with all other commands
+func NewRootCommand() *cobra.Command {
+
+	root := &cobra.Command{
+		Use:   "knit",
+		Short: "knit allows to check system settings for low-latency workload",
+
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			knitOpts.cpus, err = cpuset.Parse(knitOpts.cpuList)
+			if err != nil {
+				return fmt.Errorf("error parsing %q: %v", knitOpts.cpuList, err)
+			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Fprint(cmd.OutOrStderr(), cmd.UsageString())
```

Comment:
Would be nice a few examples with the expected output when seeing the usage

Reply:
the output could be pretty large, so I'll prepare a README.

Reply:
ok